### PR TITLE
Add webpackNodeFlags setting

### PR DESCRIPTION
**Pull request by ngbinh**
Allow users to pass https://nodejs.org/api/cli.html to webpack runs

### CODE REVIEW

File: sbt-scalajs-bundler/src/main/scala/scalajsbundler/Webpack.scala [hunk 1]
```diff
@@ -286,13 +288,14 @@ object Webpack {
...
+    * @param nodeFlags node.js cli flags
     * @param args Arguments to pass to the webpack command
     * @param workingDir Working directory in which the Nodejs will be run (where there is the `node_modules` subdirectory)
     * @param log Logger
     */
-  def run(args: String*)(workingDir: File, log: Logger): Option[WebpackStats] = {
+  def run(nodeFlags: String*)(args: String*)(workingDir: File, log: Logger): Option[WebpackStats] = {
     val webpackBin = workingDir / "node_modules" / "webpack" / "bin" / "webpack"
-    val cmd = Seq("node", webpackBin.absolutePath, "--bail", "--profile", "--json") ++ args
+    val cmd = Seq("node") ++ nodeFlags ++ Seq(webpackBin.absolutePath, "--bail", "--profile", "--json") ++ args
```
Comment (julienrf) [inline sbt-scalajs-bundler/src/main/scala/scalajsbundler/Webpack.scala]: We should then simplify the expression and not create a singleton sequence containing `"node"`.

File: sbt-scalajs-bundler/src/main/scala/scalajsbundler/sbtplugin/ScalaJSBundlerPlugin.scala [hunk 2]
```diff
@@ -348,6 +348,18 @@ object ScalaJSBundlerPlugin extends AutoPlugin {
...
     )
 
+    /**
+      * node.js cli flags https://nodejs.org/api/cli.html
+      *
+      * Defaults to an empty list.
+      *
+      * @group settings
+      */
+    val webpackNodeFlags = SettingKey[Seq[String]](
```
Comment (julienrf) [inline sbt-scalajs-bundler/src/main/scala/scalajsbundler/sbtplugin/ScalaJSBundlerPlugin.scala]: I think it should be named `webpackNodeArgs`, unless `node` uses a special terminology.

File: sbt-scalajs-bundler/src/main/scala/scalajsbundler/sbtplugin/ScalaJSBundlerPlugin.scala [hunk 3]
```diff
@@ -348,6 +348,18 @@ object ScalaJSBundlerPlugin extends AutoPlugin {
...
+      "webpackNodeFlags",
+      "Custom arguments to webpack"
```
Comment (julienrf) [inline sbt-scalajs-bundler/src/main/scala/scalajsbundler/sbtplugin/ScalaJSBundlerPlugin.scala]: The description is not correct: it’s arguments to the `node` program when webpack is run.

**Comment by ngbinh** [comment]
update with comments addressed.

**Comment by ngbinh** [comment]
Sorry, I may not be clear on the use case. We need to increase node.js memory on our scala.js bundle build as `weback` would choke from time to time on default memory setting.

File: sbt-scalajs-bundler/src/main/scala/scalajsbundler/sbtplugin/ScalaJSBundlerPlugin.scala [hunk 4]
```diff
@@ -512,7 +524,8 @@ object ScalaJSBundlerPlugin extends AutoPlugin {
     // API user can modify it just once.
     webpackMonitoredDirectories := Seq(),
     (includeFilter in webpackMonitoredFiles) := AllPassFilter,
-    webpackExtraArgs := Seq(),
+    webpackExtraArgs := Seq.empty,
```
Comment (cquiroz) [inline sbt-scalajs-bundler/src/main/scala/scalajsbundler/sbtplugin/ScalaJSBundlerPlugin.scala]: So many args :D can't find a good way around though

**Comment by julienrf** [comment]
Thanks @ngbinh! In the future, please avoid changing the code style, it makes it harder to review your work.

**Pull request merged**